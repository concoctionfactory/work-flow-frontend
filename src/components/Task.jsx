import React, { useState } from "react";
import TaskFrom from "./TaskForm";
import moment from "moment";
import { makeStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import EditIcon from "@material-ui/icons/Edit";
import Chip from "@material-ui/core/Chip";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardHeader from "@material-ui/core/CardHeader";
import Typography from "@material-ui/core/Typography";
const useStyles = makeStyles((theme) => ({
  inline: {
    display: "inline",
  },
  card: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
  },
}));
function Task({ card, boardId }) {
  const classes = useStyles();
  const { name, text, due_date, status, list_id } = card;

  const [taskForm, setTaskForm] = useState({
    isOpen: false,
  });

  const openTaskFrom = () => {
    setTaskForm({ isOpen: true });
  };

  const closeTaskFrom = () => {
    setTaskForm({ isOpen: false });
  };

  return (
    <Card className={classes.card}>
      <CardHeader
        action={
          <IconButton aria-label="edit" onClick={openTaskFrom}>
            <EditIcon />
          </IconButton>
        }
        title={
          <Typography gutterBottom="false" variant="subtitle">
            {name}
          </Typography>
        }
        subheader={
          <Typography variant="subtitle2" color="textSecondary">
            {text}
          </Typography>
        }
      />
      <CardActions>
        <Chip label={status} size="small" />
        {due_date && (
          <Chip
            label={moment(due_date, moment.defaultFormat).format(`ll`)}
            size="small"
          />
        )}
      </CardActions>

      <TaskFrom
        className={classes.fab}
        isOpen={taskForm.isOpen}
        handleOpen={openTaskFrom}
        handleClose={closeTaskFrom}
        listTaskId={list_id}
        boardId={boardId}
        task={card}
      ></TaskFrom>
    </Card>
  );
}

export default Task;
