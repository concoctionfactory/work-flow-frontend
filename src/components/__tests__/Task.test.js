import React from "react";
import { render } from "@testing-library/react";
import Task from "../Task";
import { userData, boardData } from "./testData";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "../../reducers/root";
import { BrowserRouter } from "react-router-dom";
import thunk from "redux-thunk";
const store = createStore(rootReducer, applyMiddleware(thunk));
const boardId = boardData.id;
const list = boardData.lists[0];

const card = list.cards[0];

it("renders without crashing", function () {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Task card={card} boardId={boardId} />
      </BrowserRouter>
    </Provider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Task card={card} boardId={boardId} />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows text ", function () {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <Task card={card} boardId={boardId} />
      </BrowserRouter>
    </Provider>
  );
  expect(component.getByText("not_started")).toBeInTheDocument();
});
