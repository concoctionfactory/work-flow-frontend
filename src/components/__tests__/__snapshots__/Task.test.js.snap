// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-4 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-gutterBottom"
        >
          create wireframe
        </span>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
        >
          use figma instead of sketch
        </h6>
      </div>
      <div
        class="MuiCardHeader-action"
      >
        <button
          aria-label="edit"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <div
        class="MuiChip-root MuiChip-sizeSmall"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall"
        >
          not_started
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-sizeSmall"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall"
        >
          Oct 1, 2020
        </span>
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
